### PR TITLE
New melt solver

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by bangerth** [comment]
This is of course @rrgrove6 's work -- I'm just creating the pull request for him because his browser wouldn't let him do it :-)

Not ready to merge, but ready for review.

File: source/simulator/assembly.cc [hunk 1]
```diff
@@ -997,7 +997,7 @@ namespace aspect
                                       introspection.component_masks.velocities,
                                       constant_modes);
 
-    Mp_preconditioner.reset (new LinearAlgebra::PreconditionILU());
+    Mp_preconditioner.reset (new LinearAlgebra::PreconditionAMG());
```
Comment (tjhei) [inline source/simulator/assembly.cc]: you are not setting any options here, is that on purpose? Are the defaults for AMG good?

**Comment by tjhei** [comment]
Let's first try to reproduce your numbers before we talk about cleanup to do.

Questions (for later?):
- do we always want to use AMG or only if we have melt? Do we need to benchmark this?
- can we rename ``Mp_preconditioner`` to ``schur_preconditioner`` and ``Amg_preconditioner`` to ``velocity_block_preconditioner``?

**Comment by rrgrove6** [comment]
We didn't seem to have messed with them before, thus I suppose we looked at them in the past and decided they were fine:

TrilinosWrappers::PreconditionAMG::AdditionalData::AdditionalData	(	
const bool 	elliptic = true,
const bool 	higher_order_elements = false,
const unsigned int 	n_cycles = 1,
const bool 	w_cyle = false,
const double 	aggregation_threshold = 1e-4,
const std::vector< std::vector< bool > > & 	constant_modes = std::vector<std::vector<bool> > (0),
const unsigned int 	smoother_sweeps = 2,
const unsigned int 	smoother_overlap = 0,
const bool 	output_details = false,
const char * 	smoother_type = "Chebyshev",
const char * 	coarse_type = "Amesos-KLU" 
)

**Comment by tjhei** [comment]
the defaults look okay to me.

**Comment by tjhei** [comment]
you only managed to break 172 tests ;-)

**Comment by bangerth** [comment]
Oh, 40%! That may get close to Puckett Bucket territory!

**Comment by ljhwang** [comment]
not even close.

**Pull request closed without merging**